### DISCUSSION TITLE
Date Range Picker

### DISCUSSION THREAD

**Discussion by Bassonlinho** [heroui-inc/heroui#2887] [Q&A]
Hey I'm trying to style the content of the Date Range Picker. Other components that I use from NextUI I could style with no problem using classNames and hitting exact slot name to override the styles, but with Date Range Picker I'm unable to do so.
<img width="242" alt="Screenshot 2024-04-26 at 11 38 23" src="https://github.com/nextui-org/nextui/assets/13332888/104403cc-6d0d-4d56-bb66-484978f55442">

In particular, I want to style the selected range buttons, text color in general and remove the shadow from gridHeader.

I tried accessing calendarProps -> classNames and then hitting the classes I want since they're present there (or any other slot), but nothing changed
<img width="345" alt="Screenshot 2024-04-26 at 11 40 58" src="https://github.com/nextui-org/nextui/assets/13332888/13605949-1ad6-4476-819b-a7472a52da99">

Am I doing it the right way or?

Thanks in advance!

**Comment by wingkwong** [comment]
For shadow one, I think it's correct. 

For selected range buttons and the text color, use `cellButton` instead of `cell`.

```tsx
cellButton: [
  // default text color
  "text-red-300",
  // selected case
  "data-[selected=true]:bg-secondary",
  "data-[selected=true]:text-secondary-foreground",
  // hover case
  "data-[hover=true]:bg-secondary-50",
  "data-[hover=true]:text-secondary-400",
  // selected and hover case
  "data-[selected=true]:data-[hover=true]:bg-secondary",
  "data-[selected=true]:data-[hover=true]:text-secondary-foreground",
]
```

**Reply by Bassonlinho** [reply]
<img width="671" alt="Screenshot 2024-04-26 at 12 42 34" src="https://github.com/nextui-org/nextui/assets/13332888/0ba1cdbb-8c12-482d-a4a3-03af1f87b319">

This is code

And this is output: 
<img width="269" alt="Screenshot 2024-04-26 at 12 42 55" src="https://github.com/nextui-org/nextui/assets/13332888/f669b286-5294-49a7-9cfa-ec89b9f0a44a">
Shadow is still present and no changes to the cell button

**Reply by wingkwong** [reply]
My bad. I thought this is range calendar. I think currently classNames are not forwarded from date range picker to calendar.

**Reply by Bassonlinho** [reply]
Ah crap, any plan to forward them soon? I thought I was doing something wrong at first but the same pattern I applied to other components and it worked pretty well

**Reply by element-software** [reply]
I have really struggled with this, the component itself works fine but styling has been a nightmare!

I ended up adding the below classes into my Tailwind `custom.css` file: 

```
[data-selection-start="true"] {
  @apply bg-red-400;
  @apply rounded-r-medium;
}

[data-selection-end="true"] {
  @apply bg-red-400;
  @apply rounded-l-medium;
}

[data-range-selection="true"]:not([data-selection-start="true"]):not([data-selection-end="true"]) {
  @apply bg-red-400;
  @apply rounded-none;
}

[data-slot="cell"]:hover {
  @apply bg-red-400;
  @apply text-white;
  @apply rounded-full;
}
```

It gives the below results and includes a simple hover state.

![image](https://github.com/nextui-org/nextui/assets/81567707/fe30b5c6-f86e-4f18-a579-8bc054ed228a)

On the `<DateRangePicker />` component itself I have the below classnames, which didn't really seem to work except for the `calendarContent` one:

```
      classNames={{
        base: "w-full border",
        calendar: "w-full bg-white shadow-lg border border-default-200/60",
        calendarContent: "w-[300px] bg-red-500/10",
        popoverContent: "w-full bg-red-500/10",
        separator: "w-6 h-6 text-center text-default-500",
        input: "w-full border-none outline-none focus:border-none focus:ring-0",
      }}
```

Hope this helps. I'm really unsure why the props are present in `classNames` and `calendarProps` but they are not really doing anything, hopefully this can be solved soon!

**Reply by jrgarciadev** [reply]
Why aren't you using the theme capabilities? https://nextui.org/docs/customization/customize-theme you can override the semantic colors (primary, secondary, ...etc) and the calendar will be automatically adjusted

**Reply by element-software** [reply]
> Why aren't you using the theme capabilities? https://nextui.org/docs/customization/customize-theme you can override the semantic colors (primary, secondary, ...etc) and the calendar will be automatically adjusted

I am using the theme already but it has no effect on this component.

**Reply by jrgarciadev** [reply]
Please check this https://codesandbox.io/p/devbox/boring-swartz-8wpmlj?file=%2FApp.jsx%3A6%2C22&utm_medium=sandpack

**Comment by jrgarciadev** [comment]
@wingkwong https://github.com/nextui-org/nextui/pull/3258
